src/app/clusters/thread-border-router-management-server/thread-border-router-management-server.cpp

```diff
@@ -0,0 +1,390 @@
+{
+    if (aPath.mClusterId != ThreadBorderRouterManagement::Id)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    if (!mDelegate)
+    {
+        ChipLogError(Zcl, "Thread Border Router Management server not initialized");
+        return CHIP_ERROR_UNINITIALIZED;
+    }
+    CHIP_ERROR status = CHIP_NO_ERROR;
+    switch (aPath.mAttributeId)
+    {
+    case Globals::Attributes::FeatureMap::Id: {
+        BitFlags<Feature> featureMap;
+        status = ReadFeatureMap(featureMap);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(featureMap);
+        }
+        break;
+    }
+    case Attributes::BorderRouterName::Id: {
+        char borderRouterNameBuf[kBorderRouterNameMaxLength + 1];
+        MutableCharSpan borderRouterName(borderRouterNameBuf);
+        status = ReadBorderRouterName(borderRouterName);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(borderRouterName);
+        }
+        break;
+    }
+    case Attributes::BorderAgentID::Id: {
+        uint8_t borderAgentIDBuf[kBorderAgentIdLength];
+        MutableByteSpan borderAgentID(borderAgentIDBuf);
+        status = ReadBorderAgentID(borderAgentID);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(borderAgentID);
+        }
+        break;
+    }
+    case Attributes::ThreadVersion::Id: {
+        uint16_t threadVersion;
+        status = ReadThreadVersion(threadVersion);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(threadVersion);
+        }
+        break;
+    }
+    case Attributes::InterfaceEnabled::Id: {
+        bool interfaceEnabled;
+        status = ReadInterfaceEnabled(interfaceEnabled);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = aEncoder.Encode(interfaceEnabled);
+        }
+        break;
+    }
+    case Attributes::ActiveDatasetTimestamp::Id: {
+        Optional<uint64_t> activeDatasetTimestamp;
+        status = ReadActiveDatasetTimestamp(activeDatasetTimestamp);
+        if (status == CHIP_NO_ERROR)
+        {
+            status = activeDatasetTimestamp.HasValue() ? aEncoder.Encode(DataModel::MakeNullable(activeDatasetTimestamp.Value()))
+                                                       : aEncoder.EncodeNull();
+        }
+        break;
+    }
+    default:
+        break;
+    }
+    return status;
+}
+
+void ServerInstance::CommitSavedBreadcrumb()
+{
+    if (mBreadcrumb.HasValue())
+    {
+        GeneralCommissioning::SetBreadcrumb(mBreadcrumb.Value());
+    }
+    mBreadcrumb.ClearValue();
+}
+
+void ServerInstance::OnActivateDatasetComplete(uint32_t randomNumber, CHIP_ERROR error)
+{
+    auto commandHandleRef = std::move(mAsyncCommandHandle);
+    auto commandHandle    = commandHandleRef.Get();
+    if (commandHandle == nullptr)
+    {
+        return;
+    }
+    if (mRandomNumber != randomNumber)
```

Comment:
So why does mRandomNumber need to be random?  Can't it just be a sequence counter?